Cargo.toml

```diff
@@ -50,8 +50,21 @@
 rsconf = { git = "https://github.com/mqudsi/rsconf", branch = "master" }
 
 [lib]
-crate-type = ["staticlib"]
+crate-type = ["rlib"]
 path = "fish-rust/src/lib.rs"
+doctest = false
+
+[[bin]]
+name = "fish"
+path = "fish-rust/src/bin/fish.rs"
+
+[[bin]]
+name = "fish_indent"
+path = "fish-rust/src/bin/fish_indent.rs"
+
+[[bin]]
+name = "fish_key_reader"
```

Comment:
This introduces a few compile warnings because attributes like `#![allow(non_camel_case_types)]` from `fish-rust/src/lib.rs` apply only to the lib crate.
I don't think there is a way to share those attributes across crates.
It would be good to fix either by copying the attributes to the `fish_*.rs` files or fixing the underlying problems.

Reply:
> I don't think there is a way to share those attributes across crates.

There is, actually, thanks to the relatively new [`[lints]` table](https://doc.rust-lang.org/cargo/reference/manifest.html#the-lints-section).

Reply:
great. I guess we don't want to bump the version yet

Reply:
Would adding that section break *building* with older versions?

Our lint-CI already runs stable, and any one of us will presumably have >=1.74.

So the only thing left would be warnings for our other CI and people who actually need an older release - and even in the C++ version people with older compilers would get warnings.